src/components/NewDatePicker/index.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import {Animated, View} from 'react-native';
+import moment from 'moment';
+import _ from 'underscore';
+import TextInput from '../TextInput';
+import CalendarPicker from '../CalendarPicker';
+import CONST from '../../CONST';
+import styles from '../../styles/styles';
+import {propTypes, defaultProps} from './datepickerPropTypes';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../withWindowDimensions';
+import withNavigation from '../withNavigation';
+import compose from '../../libs/compose';
+import './styles.css';
+
+const datePickerPropTypes = {
+    ...propTypes,
+    ...windowDimensionsPropTypes,
+};
+
+class NewDatePicker extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedDate: null,
+            isPickerVisible: false,
+        };
+
+        this.setDate = this.setDate.bind(this);
+        this.togglePicker = this.togglePicker.bind(this);
+        this.removeClickListener = this.removeClickListener.bind(this);
+        this.handleOutsideClick = this.handleOutsideClick.bind(this);
+
+        this.opacity = new Animated.Value(0);
+        this.wrapperRef = React.createRef();
+
+        /* We're using uncontrolled input otherwise it wont be possible to
+        * raise change events with a date value - each change will produce a date
+        * and make us reset the text input */
+        this.defaultValue = props.defaultValue
+            ? moment(props.defaultValue).format(CONST.DATE.MOMENT_FORMAT_STRING)
+            : '';
+
+        this.minDate = props.minDate ? moment(props.minDate).toDate() : null;
+        this.maxDate = props.maxDate ? moment(props.maxDate).toDate() : null;
+    }
+
+    componentDidMount() {
+        document.addEventListener('mousedown', this.handleOutsideClick);
```

Comment:
Won't clicking outside blur the picker? (assuming the picker is a focusable element)

Reply:
blur is called while navigating to YearPickerPage

Reply:
What I meant is instead of using the hacky `handleOutsideClick`, use the `onBlur` callback of the `TextInput` to hide the picker.

Reply:
I tried playing around that a bit, the downside of using `onBlur` here is that after going back from the year picker screen, the input is not focused, so it is impossible to close it clicking outside of the input. In that case we would probably have to focus the input again using its ref. When tracking the `mousedown` event, it is still possible to close the popover even if input is not focused. WDYT? I might be still missing something here, but i think tracking the `mousedown` event helps us to keep the consistency of behavior

Reply:
i have noticed another unwanted behaviour, if you click on the picker the date text input will be blurred. That should not happen, as long as you are using the picker the date picker should remain focused (as it's the case with the native date input). I think we should tackle this before discussing the removal of `handleOutsideClick`.

Reply:
what is the simple way to get around this one?

Reply:
1. Persist focus on TextInput while clicking on the calendar
    - I think we can achieve this by preventing default behaviour of `mousedown` of the `Animated.View` wrapping the `CalendarPicker`.
 2. Remove `handleOutsideClick`
 3. Remove `onYearPressed`
 4. Use the `onBlur` callback of the TextInput to hide the picker
 5. On `componentDidUpdate` if the year changed
     1. Open the picker
     2. Focus the TextInput
 6. Instead of using `togglePicker` use two functions `openPicker` and `closePicker` 
 
This should do it

Reply:
@s77rt about the first point, I'm not sure how I can achieve persist focus while clicking on the calendar using `mousedown`

Reply:
@s77rt please check my last commit